src/mrpro/operators/_PadOp.py

```diff
@@ -0,0 +1,103 @@
+"""Class for Pad Operator."""
+
+# Copyright 2023 Physikalisch-Technische Bundesanstalt
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+#   you may not use this file except in compliance with the License.
+#   You may obtain a copy of the License at
+#       http://www.apache.org/licenses/LICENSE-2.0
+#   Unless required by applicable law or agreed to in writing, software
+#   distributed under the License is distributed on an "AS IS" BASIS,
+#   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#   See the License for the specific language governing permissions and
+#   limitations under the License.
+
+import torch
+
+from mrpro.operators import LinearOperator
+from mrpro.utils import change_data_shape
+
+
+class PadOp(LinearOperator):
+    """Pad operator class."""
+
+    def __init__(
+        self,
+        dim: tuple[int, ...],
```

Comment:
the argument dim can be inferred from orig_shape and padded_shape by checking for the dimensions where orig_shape and padded_shape do not coincide... this maybe is also less prone to errors

Reply:
this is differnt, you might want to specify "pad along dimension 1" and not first calculate the final shape.
I would leave dim as an argument, but maybe allow None -> last len(orig_shape) dimensions>

Reply:
Yes, that was the idea. If you want to pad along dimension k2 you should not have to care about all the other dimensions (which could be changed by other operators). 

I would not use None here, because setting `dim = ()` does not do anything and so I don't find it intuitive if `dim = None` that then the last `len(orig_shape)` are padded. I think this makes more sense for `change_data_shape/pad_or_crop` and I would add it there.